File: Insertion_Sort/Insertion_Sort.cs
// Insertion sort in C#
using System;

class Insertion_Sort
{
    // Function to print an array
    static void PrintArray(int[] array, int size)
    {
        for (int i = 0; i < size; i++)
        {
            Console.Write(array[i] + " ");
        }
        Console.WriteLine();
    }

    static void InsertionSort(int[] array, int size)
    {
        for (int step = 1; step < size; step++)
        {
            int key = array[step];
            int j = step - 1;

            // Compare key with each element on the left of it until an element smaller than
            // it is found.
            // For descending order, change key < array[j] to key > array[j].
            while (key < array[j] && j >= 0)
            {
                array[j + 1] = array[j];
                --j;
            }
            array[j + 1] = key;
        }
    }

    static void Main(string[] args)
    {
        int[] data = { 9, 5, 1, 4, 3 };
        int size = data.Length;
        InsertionSort(data, size);
        Console.WriteLine("Sorted array in ascending order:");
        PrintArray(data, size);
    }
}

// This code is contributed by Ali Barzegari Dahaj

File: Insertion_Sort/Insertion_sort.py
# Insertion sort in python

def insertion_sort(arr):
    # Traverse through 1 to len(arr)
    for i in range(1, len(arr)):
        # Select the element to be inserted
        key = arr[i]
        
        # Initialize j to the index before i
        j = i - 1
        
        # Move elements of arr[0..i-1], that are greater than key,
        # to one position ahead of their current position
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        
        # Insert the key element after the element just smaller than it
        arr[j + 1] = key
    
    return arr


#Example
data=[-2,45,0,11,-9]
print("Sorted Array in Ascending Order: ")
print(insertion_sort(data))


# This code is contributed by Erfan Shafiee
